src/Orleans/Messaging/IncomingMessageBuffer.cs

```diff
@@ -154,7 +161,11 @@ public bool TryDecodeMessage(out Message msg)
             List<ArraySegment<byte>> body = ByteArrayBuilder.BuildSegmentListWithLengthLimit(readBuffer, bodyOffset, bodyLength);
             
             // build message
-            msg = new Message(header);
+            var context = new DeserializationContext(this.serializationManager)
```

Comment:
These are new. I'll see if I can allocate these once per message buffer - depending on what the access patterns are like.

Agree that we should consider an object pool.